src/LibUvLoop.php

```diff
@@ -0,0 +1,263 @@
+<?php
+
+namespace React\EventLoop;
+
+use React\EventLoop\Tick\FutureTickQueue;
+use React\EventLoop\Timer\Timer;
+use React\EventLoop\Timer\TimerInterface;
+use SplObjectStorage;
+
+/**
+ * @see https://github.com/bwoebi/php-uv
+ */
+class LibUvLoop implements LoopInterface
+{
+    private $uv;
+    private $futureTickQueue;
+    private $timerEvents;
+    private $events = [];
+    private $listeners = [];
+    private $running;
+    private $streamListener;
+
+    public function __construct()
+    {
+        $this->uv = \uv_loop_new();
+        $this->futureTickQueue = new FutureTickQueue();
+        $this->timerEvents = new SplObjectStorage();
+        $this->streamListener = $this->createStreamListener();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addReadStream($stream, callable $listener)
+    {
+        if (isset($this->listeners[(int) $stream]['read'])) {
+            return;
+        }
+
+        $this->listeners[(int) $stream]['read'] = $listener;
+        $this->addStream($stream);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addWriteStream($stream, callable $listener)
+    {
+        if (isset($this->listeners[(int) $stream]['write'])) {
+            return;
+        }
+
+        $this->listeners[(int) $stream]['write'] = $listener;
+        $this->addStream($stream);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function removeReadStream($stream)
+    {
+        if (!isset($this->events[(int) $stream])) {
+            return;
+        }
+
+        unset($this->listeners[(int) $stream]['read']);
+
+        if (!isset($this->listeners[(int) $stream]['read'])
+            && !isset($this->listeners[(int) $stream]['write'])) {
+            unset($this->events[(int) $stream]);
+        }
+    }
```

Comment:
Isn't this supposed to call `uv_poll_stop()` if this stream does no longer have a listener? (see also below)

Reply:
Yup, just pushed changes that address that 👍